src/tests/ftest/datamover/dm_obj_small.py

```diff
@@ -0,0 +1,255 @@
+
+                for akey_idx in range(self.num_akeys_single_per_dkey):
+                    # Round-robin to get the size of data and
+                    # arbitrarily use a number 0-9 to fill data
+                    akey_size_idx = akey_idx % len(self.akey_sizes)
+                    data_size = self.akey_sizes[akey_size_idx]
+                    data_val = str(akey_idx % 10)
+                    data = data_size * data_val
+                    akey = "akey single {}".format(akey_idx)
+                    c_akey = ctypes.create_string_buffer(akey)
+                    buf = ioreq.single_fetch(c_dkey, c_akey, data_size + 1)
+                    actual_data = str(buf.value)
+                    if actual_data != data:
+                        self.log.info("Expected:\n%s\nBut got:\n%s",
+                            str(data[:100]) + "...",
+                            str(actual_data[:100]) + "...")
+                        self.log.info(
+                            "For:\nobj: %s.%s\ndkey: %s\nakey: %s",
+                                str(obj.c_oid.hi), str(obj.c_oid.lo),
+                                dkey, akey)
+                        self.fail("Single value verification failed.")
+
+                for akey_idx in range(self.num_akeys_array_per_dkey):
+                    # Round-robin to get the size of data and
+                    # the number of extents, and
+                    # arbitrarily use a number 0-9 to fill data
+                    akey_size_idx = akey_idx % len(self.akey_sizes)
+                    data_size = self.akey_sizes[akey_size_idx]
+                    c_data_size = ctypes.c_size_t(data_size)
+                    akey_extent_idx = akey_idx % len(self.akey_extents)
+                    num_extents = self.akey_extents[akey_extent_idx]
+                    c_num_extents = ctypes.c_uint(num_extents)
+                    akey = "akey array {}".format(akey_idx)
+                    c_akey = ctypes.create_string_buffer(akey)
+                    actual_data = ioreq.fetch_array(c_dkey, c_akey,
+                                                    c_num_extents, c_data_size)
+                    for data_idx in range(num_extents):
+                        data_val = str(data_idx % 10)
+                        data = data_size * data_val
+                        if data != actual_data[data_idx]:
+                            self.log.info("Expected:\n%s\nBut got:\n%s",
+                                str(data[:100]) + "...",
+                                str(actual_data[data_idx][:100]) + "...")
+                            self.log.info(
+                                "For:\nobj: %s.%s\ndkey: %s\nakey: %s",
+                                    str(obj.c_oid.hi), str(obj.c_oid.lo),
+                                    dkey, akey)
+                            self.fail("Array verification failed.")
+            obj.close()
+        cont.close()
+
+    def run_dm_obj_small(self, tool):
```

Comment:
It would be great if you could add the description for `tool`.

Reply:
Done!